doc/development/guide/tests.rst

```diff
@@ -1,32 +1,56 @@
 Software tests
 ==============
 
-Running the test suite
-~~~~~~~~~~~~~~~~~~~~~~
+Running the tests
+~~~~~~~~~~~~~~~~~
 
-The PennyLane test suite requires the Python ``pytest`` package, as well as ``pytest-cov``
-for test coverage, ``pytest-mock`` for mocking, and ``flaky`` for automatically rerunning flaky tests; these can be installed via ``pip``:
+The PennyLane test suite requires the Python ``pytest`` package, as well as:
+
+* ``pytest-cov``: determines test coverage
+* ``pytest-mock``: allows replacing components with dummy objects
+* ``flaky``: manages tests with non-deterministic behaviour
+
+ These requirements can be installed via ``pip``:
 
 .. code-block:: bash
 
     pip install pytest pytest-cov pytest-mock flaky
 
-To ensure that PennyLane is working correctly, the test suite can be run by
-navigating to the source code folder and running
+The `tests <https://github.com/PennyLaneAI/pennylane/tree/master/tests>`__ folder of the root PennyLane directory contains the PennyLane test suite. Run all tests in this folder via:
 
 .. code-block:: bash
 
-    make test
+    python -m pytest --ignore=tests/beta tests
 
-while the test coverage can be checked by running
+Using ``python -m`` ensures that the tests run with the correct Python version if multiple versions are on the system. The `tests/beta` folder can contain failing tests, so ``--ignore=tests/beta`` excludes them from execution.
+
+As the entire test suite takes some time, locally running only relevant files speeds the debugging cycle.  For example, if a developer was adding a new non-parametric operation, they could run:
 
 .. code-block:: bash
 
-    make coverage
+    python -m pytest tests/ops/qubit/test_non_parametric_ops.py
 
-The output of the above command will show the coverage percentage of each
-file, as well as the line numbers of any lines missing test coverage.
+The slowest tests are marked with ``slow`` and can be deselected by:
 
+.. code-block:: bash
+
+    python -m pytest -m "not slow" tests
+
+Pytest supports many other command-line options, which can be found with the command:
+
+.. code-block:: bash
+
+    pytest --help
+
+Or by visiting the `pytest documentation <https://docs.pytest.org/en/latest/reference/reference.html#id88>`__ . 
+
+PennyLane provides a set of tests for all PennyLane plugins. See the documentation on these tests under the section on the `device API <https://pennylane.readthedocs.io/en/latest/code/api/pennylane.devices.tests.html>`__.
```

Comment:
Maybe we could add the example at https://pennylane.readthedocs.io/en/latest/code/api/pennylane.devices.tests.html#the-pl-device-test-cli in this page too?

That could give a rough idea of how the device test suite works and interested readers could then redirect to the other page. Otherwise, I'd think some ppl. might just skip looking there in the first place :sweat_smile: